### PR TITLE
Implement azure artifact store and fileio plugin

### PR DESCRIPTION
**Pull request by schustmi**
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe changes
- Implement azure artifact store and fileio plugin

### CODE REVIEW

File: src/zenml/integrations/azure/__init__.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+from zenml.integrations.constants import AZURE
+from zenml.integrations.integration import Integration
+
+
+class AzureIntegration(Integration):
+    """Definition of Azure integration for ZenML."""
+
+    NAME = AZURE
+    REQUIREMENTS = ["adlfs"]
```
Comment (htahir1) [inline src/zenml/integrations/azure/__init__.py:28]: Maybe we can start pinning these?

**Review by htahir1 — COMMENTED** [review]
Nice what a cool addition!

File: src/zenml/integrations/azure/__init__.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""
+The Azure integration submodule provides a way to run ZenML pipelines in a cloud
+environment. Specifically, it allows the use of cloud artifact stores,
+and an `io` module to handle file operations on Azure Data Lakes .
```
Comment (wjayesh) [inline src/zenml/integrations/azure/__init__.py:17]: We only added support for Azure Blob Storage, as of now.

File: src/zenml/integrations/azure/io/azure_plugin.py [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+from tfx.dsl.io.fileio import NotFoundError
+
+from zenml.io.fileio import convert_to_str
+from zenml.io.filesystem import Filesystem, PathType
+
+
+class ZenAzure(Filesystem):
+    """Filesystem that delegates to Azure storage using adlfs.
+
+    To authenticate with an Azure Data Lake storage, make sure to set a
```
Comment (wjayesh) [inline src/zenml/integrations/azure/io/azure_plugin.py:32]: Small correction from Data Lake to Blob Storage

**Review by wjayesh — CHANGES_REQUESTED** [review]

**Review by wjayesh — APPROVED** [review]

**Pull request merged**